dbms/src/Interpreters/Join.cpp

```diff
@@ -2301,4 +2370,20 @@ BlockInputStreamPtr Join::createStreamWithNonJoinedRows(const Block & left_sampl
     return std::make_shared<NonJoinedBlockInputStream>(*this, left_sample_block, index, step, max_block_size);
 }
 
+void ProbeProcessInfo::resetBlock(Block && block_)
+{
+    block = std::move(block_);
+    start_row = 0;
+    end_row = 0;
+    all_rows_joined_finish = true;
```

Comment:
why set `all_rows_joined_finish` to true here?

Reply:
> why set `all_rows_joined_finish` to true here?

It should be initialized to true，or the hash join probe stream read can not get new block in first time.

Reply:
move to constructor.